lib/mutant/ast/regexp.rb

```diff
@@ -0,0 +1,41 @@
+module Mutant
+  module AST
+    # Regexp source mapper
+    module Regexp
+      UNSUPPORTED_EXPRESSION_TYPE = :conditional
+      private_constant(*constants(false))
+
+      # Check if expression is supported by mapper
+      #
+      # @param expression [Regexp::Expression]
+      #
+      # @return [Boolean]
+      def self.supported?(expression)
+        return if expression.type.equal?(UNSUPPORTED_EXPRESSION_TYPE)
```

Comment:
This method does always return `false` when the method isn't being called recursively. If the method is being called with an expression from anywhere else then it is likely being called with a `Regexp::Expression::Root` which will never be `UNSUPPORTED_EXPRESSION_TYPE` and will never be `terminal?`